File: ViewBase/Source/Converters/InverseBooleanConverter.vb
Imports System.Globalization
Imports System.Windows.Data

Public Class InverseBooleanConverter
    Implements IValueConverter

    Public Function Convert(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.Convert
        If value.GetType Is GetType(Boolean) Then
            Return Not (CType(value, Boolean))
        End If
        Return False
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
        If value.GetType Is GetType(Boolean) Then
            Return Not (CType(value, Boolean))
        End If
        Return False
    End Function
End Class

File: BaseView/src/Converters/InverseBooleanConverter.cs
using System;
using System.Globalization;
using System.Windows.Data;

namespace Wells.BaseView.Converters
{
    public class InverseBooleanConverter : IValueConverter
    {
        object IValueConverter.Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            if (value.GetType() == typeof(bool)) { return !(bool)value; }
            return false;
        }

        object IValueConverter.ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            if (value.GetType() == typeof(bool)) { return !(bool)value; }
            return false;
        }
    }
}
